### DISCUSSION TITLE
Next js 12.1.0 fails to serve static images

### DISCUSSION THREAD

**Discussion by samuelhulla** [vercel/next.js#34933] [Help] [answered]
### Verify canary release

- [X] I verified that the issue exists in Next.js canary release

### Provide environment information

```bash
npx --no-install next info                                                  ✔  29s   17:45:37  
/bin/sh: pnpm: command not found

    Operating System:
      Platform: darwin
      Arch: arm64
      Version: Darwin Kernel Version 21.3.0: Wed Jan  5 21:37:58 PST 2022; root:xnu-8019.80.24~20/RELEASE_ARM64_T8101
    Binaries:
      Node: 16.14.0
      npm: 8.3.1
      Yarn: 1.22.17
      pnpm: N/A
    Relevant packages:
      next: 12.1.1-canary.4
      react: 18.0.0-rc.0
      react-dom: 18.0.0-rc.0

(node:3679) [DEP0148] DeprecationWarning: Use of deprecated folder mapping "./" in the "exports" field module resolution of the package at /Users/samuel/Coding/portfolio/node_modules/react/package.json.
Update this package.json to use a subpath pattern like "./*".
(Use `node --trace-deprecation ...` to show where the warning was created)
```

### How are you deploying your application? (if relevant)

Vercel

### Describe the Bug

Hello, my static images just flat out stopped displaying after a while (again), after updating from `Next 12.0` to `Next 12.1` 

Consider the following file structure

```bash
 - public/
    - parallax/
       - background.png
       -  ...
 - components/
    - Parallax.client.tsx 
```

and pretty standard stuff when it comes to  service the image

```ts
// Paralax.client.tsx
import Image from 'next/image'
import background from '../public/parallax/background.png'

const Parallax = () => (
  // obviously there's more here in actual code, but for example clarity sake...
  <Image layout="responsive" src={backgorund} alt="Background" />
)
```

### Expected Behavior

I'd expect the static image to be fetched properly (as it previously did), but instead of an actual url, inspecting the image returns instead 

`data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7`

which just points to blank image

### To Reproduce

Install Nextjs 12.1.0 and use a static image (?) 

Reproducing in code-sandbox however did not succeed as it appears to have worked properly... Perhaps it may be some weird caching issue?

**Comment by balazsorban44** [comment]
Why do you have components under the `public` folder? :thinking: Adding a repository link would help in any case.

**Reply by samuelhulla** [reply]
@balazsorban44  Apologies, that was just formatting error on my part. components are in the project root.

**Comment by samuelhulla** [comment, marked as answer]
Alright, I spent actually a lot of hours tinkering around with next config and individual files.

I found out what the issue was - given I was previously using the Next.js 12.0 build, in order to opt-in to the concurrent features, you had to add

```js
// next.config.js
module.exports = {
  experimental: {
     concurrentFeatures: true,
  }
  // ...
}
```

In order to enable them. However in Next.js 12.1 this has been deperecated as pointed out [here](https://github.com/vercel/next.js/pull/34068) and should now be replaced with `runtime` experimental flag.

> Nextjs on running dev does technically throw an error if you have concurrentFeatures enabled that a `runtime` flag is now required, however it still allows you to run the next application without removing the `concurrentFeatures` flag, which was cuasing issues with the build.

Changing your `next.config.js` to

```js
// next.config.js
module.exports = {
  experimental: {
     runtime: 'edge', // or runtime: 'nodejs'
  }
  // ...
}
```

fixes the issue, that being said, there should probably be a warning raised when these conflicting flags exist or at the very least this should be listed as a breaking change in changelog from 12.0 to 12.1
